### PR TITLE
Rewrite testgrid config upload use bazel genrule

### PR DESCRIPTION
**Pull request by krzyzacy**
/assign @ixdy 

Thanks and seems we are moving towards a better world!

### CODE REVIEW

File: testgrid/config-upload.sh [hunk 1]
```diff
@@ -18,18 +18,5 @@ set -o nounset
...
-
-# Export config
-export CONFIGDIR=$TEST_INFRA/testgrid/config
-go run $CONFIGDIR/main.go --yaml=$CONFIGDIR/config.yaml --output=$CONFIGDIR/config
-gsutil cp $CONFIGDIR/config gs://k8s-testgrid/config
-rm $CONFIGDIR/config
-
-# Remove duplicated build
-rm -rf $TEST_INFRA
+bazel build //testgrid/config:gen-config
```
Comment (ixdy) [inline testgrid/config-upload.sh]: nit: you can also use the output name here, which would be more consistent with the next line.

i.e. `bazel build //testgrid/config:testgrid-config` should work.

Comment (ixdy) [inline testgrid/config-upload.sh]: also, it looks like this currently runs on jenkins, where we don't have bazel installed. maybe this needs to be moved to prow w/ bootstrap and the bazelbuild image?

Comment (krzyzacy) [inline testgrid/config-upload.sh]: this is running from prow :-)

Comment (krzyzacy) [inline testgrid/config-upload.sh]: https://github.com/kubernetes/test-infra/blob/master/prow/config.yaml#L1206

Comment (ixdy) [inline testgrid/config-upload.sh]: so it is!

need to switch to the bazel image, though.

Comment (krzyzacy) [inline testgrid/config-upload.sh]: ah I see you can bazel build an output file directly. I made them have the same name and getting `target 'testgrid-config' is both a rule and a file; please choose another name for the rule`, now updated :-)

**Comment by krzyzacy** [comment]
Using `:0.12` for now.. https://github.com/kubernetes/test-infra/pull/3630 is not in yet

File: prow/config.yaml [hunk 2]
```diff
@@ -1208,7 +1208,7 @@ postsubmits:
     - master
     spec:
       containers:
-      - image: gcr.io/k8s-testimages/gcloud-in-go:0.2
+      - image: gcr.io/k8s-testimages/bazelbuild:0.12
         args:
         - "--branch=$(PULL_REFS)"
```
Comment (ixdy) [inline prow/config.yaml]: with 0.12+ you need to remove this.

with @fejta's PR you need to have `--repo`.

0.13 now exists (with bazel 0.5.2).

Comment (krzyzacy) [inline prow/config.yaml]: 0.13 it is!

**Comment by ixdy** [comment]
/lgtm

depending on whether you or @fejta wins the race, someone gets to rebase here :)

**Comment by fejta** [comment]
I'm happy to rebase

**Comment by krzyzacy** [comment]
prow's not reporting status back?

**Comment by krzyzacy** [comment]
hummm I got 
`W0730 23:52:49.376] ____Loading package: @io_bazel_rules_go_toolchain//
W0730 23:52:49.548] ERROR: /workspace/k8s.io/test-infra/BUILD:7:12: Label '//:.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/local_jdk/jre/bin/rmiregistry' crosses boundary of subpackage '.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/local_jdk' (perhaps you meant to put the colon here: '//.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/external/local_jdk:jre/bin/rmiregistry'?).
`

**Comment by ixdy** [comment]
This is probably another manifestation of https://github.com/kubernetes/kubernetes/issues/49844.

wonder if there is a good way we can do a `bazel clean` across everything, or at least wipe out the existing bazel cache PDs...

**Comment by krzyzacy** [comment]
ah, k/k and test-infra shares the same workspace? So I need to make a `bazel clean` call before `bazel build` here?

**Pull request merged**